internal/worker/rundynamic.go

```diff
@@ -14,35 +14,85 @@ import (
-provided, across all phases (e.g. import, install) valid in the package ecosystem.
-Status and errors are logged to stdout. There are 4 return values:
+DynamicAnalysisResult holds all the results from RunDynamicAnalysis
 
-DynamicAnalysisResults: Map of each successfully run phase to a summary of
+Data: Map of each successfully run phase to a summary of
 the corresponding dynamic analysis result. This summary has two parts:
 1. StraceSummary: information about system calls performed by the process
 2. FileWrites: list of files which were written to and counts of bytes written
 
 Note, if error is not nil, then results[lastRunPhase] is nil.
 
-RunPhase: the last phase that was run. If error is non-nil, this phase did not
+LastRunPhase: the last phase that was run. If error is non-nil, this phase did not
 successfully complete, and the results for this phase are not recorded.
 Otherwise, the results contain data for this phase, even in cases where the
 sandboxed process terminated abnormally.
 
 Status: the status of the last run phase if it completed without error, else empty
-
-error: Any error that occurred in the runtime/sandbox infrastructure.
-This does not include errors caused by the package under analysis.
 */
 
-func RunDynamicAnalysis(pkg *pkgmanager.Pkg, sbOpts []sandbox.Option) (analysisrun.DynamicAnalysisResults, analysisrun.DynamicPhase, analysis.Status, error) {
+type DynamicAnalysisResult struct {
+	Data         analysisrun.DynamicAnalysisResults
+	LastRunPhase analysisrun.DynamicPhase
+	LastStatus   analysis.Status
+}
+
+// defaultDynamicAnalysisImage is container image name of the default dynamic analysis sandbox
+const defaultDynamicAnalysisImage = "gcr.io/ossf-malware-analysis/dynamic-analysis"
+
+// defaultDynamicAnalysisCommand returns the path (in the default sandbox image)
+// of the default dynamic analysis command for the ecosystem
+var defaultDynamicAnalysisCommand = map[pkgecosystem.Ecosystem]string{
+	pkgecosystem.CratesIO:  "/usr/local/bin/analyze-rust.py",
+	pkgecosystem.NPM:       "/usr/local/bin/analyze-node.js",
+	pkgecosystem.Packagist: "/usr/local/bin/analyze-php.php",
+	pkgecosystem.PyPI:      "/usr/local/bin/analyze-python.py",
+	pkgecosystem.RubyGems:  "/usr/local/bin/analyze-ruby.rb",
+}
+
+// execArgs returns the arguments to pass to the dynamic analysis command in the sandbox
+// for the given phase of dynamic analysis on a package. The actual analysis command
+// depends on the ecosystem, see pkgmanager.PkgManager.DynamicAnalysisCommand()
+func analysisArgs(p *pkgmanager.Pkg, phase analysisrun.DynamicPhase) []string {
+	args := make([]string, 0)
+
+	if p.IsLocal() {
+		args = append(args, "--local", p.LocalPath())
+	} else if p.Version() != "" {
+		args = append(args, "--version", p.Version())
+	}
+
+	if phase == "" {
+		args = append(args, "all")
+	} else {
+		args = append(args, string(phase))
+	}
+
+	args = append(args, p.Name())
+
+	return args
+}
```

Comment:
Long term I'm not sure this is the right place, but I'm okay with the general direction of making pkgmanager more about the interaction with the Package Repository APIs.

In the short term - maybe this better belongs in the dynamicanalysis package?